### PR TITLE
[io-kafka] Position.OLDEST should use last committed offset when usin…

### PR DESCRIPTION
**Pull request by dmvk**
…g named consumer.

### CODE REVIEW

File: direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java [hunk 1]
```diff
@@ -648,14 +649,24 @@ public void waitUntilReady() throws InterruptedException {
...
-          consumer.poll(Duration.ofMillis(accessor.getAssignmentTimeoutMillis()));
+          emptyPoll =
+              consumer.poll(Duration.ofMillis(accessor.getAssignmentTimeoutMillis())).isEmpty();
+        }
+        final Set<TopicPartition> assignment = consumer.assignment();
+        final Map<TopicPartition, OffsetAndMetadata> committed = consumer.committed(assignment);
+        if (committed.values().stream().noneMatch(Objects::nonNull)) {
+          log.info("Seeking consumer name {} to beginning of partitions {}", name, assignment);
+          consumer.seekToBeginning(assignment);
+        } else {
```
Comment (je-ik) [inline direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java:664]: `else if`?

File: direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/Consumers.java [hunk 2]
```diff
@@ -119,7 +120,7 @@ public boolean consumeWithConfirm(
       }
       committed.compute(tp.partition(), (k, v) -> v == null || v <= offset ? offset + 1 : v);
       committer.confirm(tp, offset);
-      return true;
+      return watermark < Watermarks.MAX_WATERMARK;
```
Comment (je-ik) [inline direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/Consumers.java:123]: We must move this out of this PR or add a test for that.

File: direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java [hunk 3]
```diff
@@ -648,14 +649,24 @@ public void waitUntilReady() throws InterruptedException {
...
+          if (!emptyPoll) {
+            log.info("Seeking consumer name {} to committed offsets {}", name, committed);
```
Comment (je-ik) [inline direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java:666]: This branch looks untested, can we fix that?

**Review by je-ik — CHANGES_REQUESTED** [review]

File: direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java (hunk 3 above)
Comment (dmvk) [reply direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java:666]: this is cover by Kafka IT

File: direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/Consumers.java (hunk 2 above)
Comment (dmvk) [reply direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/Consumers.java:123]: Also cover in Kafka IT

**Review by je-ik — APPROVED** [review]
LGTM, thanks!

**Pull request merged**